workflow/controller/entrypoint/container_registry_index.go

```diff
@@ -27,7 +29,11 @@ func (i *containerRegistryIndex) Lookup(ctx context.Context, image string, optio
 	if err != nil {
 		return nil, err
 	}
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(kc))
+	var defaultPlatform = pkgv1.Platform{
+		Architecture: runtime.GOARCH,
```

Comment:
Note to other reviewers: I wasn't sure if there was a one-to-one correspondence between `GOARCH` and the OCI `architecture` field, so I checked [the spec](https://github.com/opencontainers/image-spec/blob/main/image-index.md), and it says:
> Image indexes SHOULD use, and implementations SHOULD understand, values listed in the Go Language document for [GOARCH](https://golang.org/doc/install/source#environment).